suite/tests/linux/clone.c

```diff
@@ -215,19 +220,19 @@ create_thread(int (*fcn)(void *), void *arg, void **stack, bool share_sighand)
  * its own.
  */
 int
-make_clone3_syscall(struct clone_args *clone_args, void (*fcn)(void))
+make_clone3_syscall(struct clone_args *clone_args, uint clone_args_size,
+                    void (*fcn)(void))
 {
-#    ifdef X86
     int ret;
+#    ifdef X86
 #        ifdef X64
     asm volatile("syscall\n\t"
                  "test %%rax, %%rax\n\t"
                  "jnz parent\n\t"
                  "call *%%rdx\n\t"
                  "parent:\n\t"
                  : "=a"(ret)
-                 : "0"(SYS_clone3), "D"(clone_args), "S"(sizeof(struct clone_args)),
-                   "d"(fcn)
+                 : "0"(SYS_clone3), "D"(clone_args), "S"(clone_args_size), "d"(fcn)
```

Comment:
(I never learned all these codes: I assume they are loading into specific registers up front.  Personally I find explicit setup and named parameters the easiest to read like at https://github.com/DynamoRIO/dynamorio/blob/master/suite/tests/linux/rseq.c#L184)

Reply:
Done.